pandas/io/tests/test_sql.py

```diff
@@ -918,6 +918,25 @@ def test_sqlalchemy_type_mapping(self):
         table = sql.SQLTable("test_type", db, frame=df)
         self.assertTrue(isinstance(table.table.c['time'].type, sqltypes.DateTime))
 
+    def test_to_sql_read_sql_with_database_uri(self):
+
+        # Test read_sql and .to_sql method with a database URI (GH10654)
+        from pandas import read_csv
+        iris_csv_file = os.path.join(tm.get_data_path(), 'iris.csv')
+        iris_frame1 = read_csv(iris_csv_file)
```

Comment:
I think this is not really needed, you can just make a small dummy frame to test yourself, or maybe better use `self.test_frame1`